pkg/docker/docker.go

```diff
@@ -241,6 +241,93 @@ func (c *Client) SupportedPlatforms(ctx context.Context) ([]v1.Platform, error)
 	}, nil
 }
 
+// ImageDistribution fetches the details for the specified image by asking
+// docker to fetch the distribution manifest from the registry. The digest
+// retrieved may not appear accurate when compared to the hub website but
+// this is expected.
+//
+// cf:
+//   - https://github.com/moby/moby/issues/40636)
+//   - https://github.com/docker/roadmap/issues/262
+//
+// When a docker image is available on only a single platform, the digest
+// shown will be the digest pointing directly at the manifest for that image
+// on that platform (as shown by the docker hub).  Where multiple platforms
+// are available, the digest is pointing to a top level document describing
+// all of the different platform manifests available.
+//
+// In either case, `docker pull` will do the correct thing and download the
+// image for your platform. For example:
+//
+// $ docker manifest inspect  bitnami/rabbitmq@sha256:0be0d2a2 ...
+//
+//	"manifests": [ {
+//		  "digest": "sha256:959a02013e8ab5538167f9....",
+//		  "platform": { "architecture": "amd64", "os": "linux" }
+//		},
+//		{
+//		  "digest": "sha256:11ee2c7e9e69e3a8311a19....",
+//		  "platform": { "architecture": "arm64", "os": "linux"}
+//		}]
+//
+// $ docker pull bitnami/rabbitmq@sha256:0be0d2a2 ...
+// $ docker image ls
+// bitnami/rabbitmq ... 48603925e10c
+//
+// The digest 486039 can be found in manifest sha256:11ee2c7e which is the manifest for
+// the current authors machine.
+//
+// $ docker manifest inspect bitnami/rabbitmq@sha256:11ee2c7e
+//
+//	  "config": {
+//		   "size": 7383,
+//		   "digest": "sha256:48603925e10c01936ea4258f...."
+//	  }
+//
+// This is the image that will finally be installed.
+func (c *Client) ImageDistribution(
+	ctx context.Context, image string,
+	forceRemote bool, creds config.DockerCredentials,
+) (*ImageManifest, error) {
+	// Check the local repository first to see if we already have this image and if so,
+	// return those details, but only if the caller did not set forceRemote
+	if !forceRemote {
```

Comment:
I can only see one place this function is used, and we don't set `forceRemote`. So is this bool necessary? Does this mean calls to this method always rely on a cached value if one exists, regardless of the cached time?

Perhaps I could do with a comment explaining the overall flow here – i.e. when does the image resolver and it's cache come into play, and how does that cache interact with this (implicit) one?

Do we also need to update our use of `ImagePlatforms` used by the bidding strategy to use the caches? https://github.com/bacalhau-project/bacalhau/blob/8d752fcef94c2048dbdf4d3bcfa77cfa0d445a09/pkg/docker/docker.go#L213

Reply:
forceRemote was just a useful tool for manual testing of remote requests, will remove and update with a comment explaining how the flow works in the resolver. 

>Do we also need to update our use of ImagePlatforms used by the bidding strategy to use the caches? 
No, this is done in the compute node isn't it? The code in the PR is in the requester.